src/clue.js

```diff
@@ -2,25 +2,119 @@
+    {firstName: "Doctor",
+    lastName: "Orchid",
+    occupation: "Scientist",
+    age: 26,
+    description: "PhD in plant toxicology. Adopted daughter of Mr. Boddy",
+    image: "http://www.radiotimes.com/uploads/images/Original/111967.jpg",
+    color: "white"},
+    {firstName: "Victor",
+    lastName: "Plum",
+    occupation: "Designer",
+    age: 22,
+    description: "Billionaire video game designer",
+    image: "https://66.media.tumblr.com/ee7155882178f73b3781603f0908617c/tumblr_phhxc7EhPJ1w5fh03_540.jpg",
+    color: "purple"},
+   {firstName: "Kasandra",
+    lastName: "Scarlet",
+    occupation: "Actor",
+    age: 31,
+    description: "She is an A-list movie star with a dark past",
+    image: "https://www.radiotimes.com/uploads/images/Original/111967.jpg",
+    color: "red"},
+    {firstName: "Eleanor",
+    lastName: "Peacock",
+    occupation: "Socialité",
+    age: 36,
+    description: "She is from a wealthy family and uses her status and money to earn popularity",
+    image: "https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg",
+    color: "blue"},
+   {firstName: "Jack",
+    lastName: "Mustard",
+    occupation: "Retired Football player",
+    age: 62,
+    description: "He is a former football player who tries to get by on his former glory",
+    image: "https://static.independent.co.uk/s3fs-public/thumbnails/image/2016/07/04/08/unspecified-3.jpg",
+    color: "yellow"}];
 
 // Rooms Array
 
-const roomsArray = [];
+const roomsArray = [
+    {name: "Dining Room"},
+    {name: "Conservatory"},
+    {name: "Kitchen"},
+    {name: "Study"},
+    {name: "Library"},
+    {name: "Billiard Room"},
+    {name: "Lounge"},
+    {name: "Ballroom"},
+    {name: "Hall"},
+    {name: "Spa"},
+    {name: "Living Room"},
+    {name: "Observatory"},
+    {name: "Theater"},
+    {name: "Guest House"},
+    {name: "Patio"}];
 
 // Weapons Array
 
-const weaponsArray = [];
+const weaponsArray = [{
+    name: "rope",
+    weight: 10},
+    {name: "knife",
+    weight: 8},
+    {name: "candlestick",
+    weight: 2},
+    {name: "dumbbell",
+    weight: 30},
+    {name: "poison",
+    weight: 2},
+    {name: "axe",
+    weight: 15},
+    {name: "bat",
+    weight: 13},
+    {name: "trophy",
+    weight: 25},
+    {name: "pistol",
+    weight: 20}];
+
 
 
 // ITERATION 2
 
-function selectRandom() {}
 
-function pickMystery() {}
+function selectRandom(arr) {
+
+    return arr[Math.floor(Math.random()*(arr.length))];
+}
+
+function pickMystery() {
+    let result = {
+        suspect:"",
+        weapon:"",
+        room:""
+    };
+    result.suspect = selectRandom(suspectsArray);
+    result.weapon = selectRandom(weaponsArray);
+    result.room = selectRandom(roomsArray);
+
+return result;
+}
 
 
 // ITERATION 3
 
-function revealMystery() {}
+function revealMystery(envelope) {
+    return `${envelope.suspect.firstName} ${envelope.suspect.lastName} killed Mr. Boddy using the ${envelope.weapon.name} in the ${envelope.room.name}!`
+
```

Comment:
Here, you have everything correct, however, it is good practice if you destructure your `envelope` argument.
By destructuring, it clearly expresses which properties you're using within the function.
By extracting the properties into individual variables (suspect, weapon, and room), you can directly reference them in your return statement without needing to use `envelope.suspect`, `envelope.weapon`, and `envelope.room` every time. This simplifies the code and improves readability.
It would look something like this:

```
function revealMystery(envelope) {
  let { suspect, weapon, room } = envelope;

  return `${suspect.firstName} ${suspect.lastName} killed Mr. Boddy using the ${weapon.name} in the ${room.name}!`;
}
```

Reply:
Ah i see, so basically making a variable inside the function that helps clarify at a glance what sort of property is meant to be used with it.

I did wonder how would a different dev know at a glance what sort of properties I wanted to be used with my function. Didn't even consider the obvious of just describing within it.

Thanks so much for the heads-up!